File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/dtos/responses/statistics/getSlotStatisticsResponse.ts
import { CustomEventResponse, LoadEventResponse, mapCustomEventResponseToCustomEvent, mapLoadEventResponseToLoadEvent, mapServerCustomStatisticsResponseToServerCustomStatistics, mapServerLifecycleStatisticsResponseToServerLifecycleStatistics, mapServerLoadStatisticsResponseToServerLoadStatistics, ServerCustomStatisticsResponse, ServerLifecycleStatisticsResponse, ServerLoadStatisticsResponse, SlotStatistics } from "../../../..";

export interface GetSlotStatisticsResponse {
    collectedDateUTC: Date,
    generalStatistics: ServerLifecycleStatisticsResponse | null,
    loadStatistics: ServerLoadStatisticsResponse | null,
    customEventStatistics: ServerCustomStatisticsResponse | null,
    lastLoadEvents: LoadEventResponse[],
    lastCustomEvents: CustomEventResponse[],
}

export function mapGetSlotStatisticsResponseToSlotStatistics(response: GetSlotStatisticsResponse): SlotStatistics {
    return {
        collectedDateUTC: response?.collectedDateUTC,
        generalStatistics: response.generalStatistics === null ? null : mapServerLifecycleStatisticsResponseToServerLifecycleStatistics(response.generalStatistics),
        loadStatistics: response.loadStatistics === null ? null : mapServerLoadStatisticsResponseToServerLoadStatistics(response.loadStatistics),
        customEventStatistics: response.customEventStatistics === null ? null : mapServerCustomStatisticsResponseToServerCustomStatistics(response.customEventStatistics),
        lastLoadEvents: response.lastLoadEvents.map(mapLoadEventResponseToLoadEvent),
        lastCustomEvents: response.lastCustomEvents.map(mapCustomEventResponseToCustomEvent),
    };
}

File: src/ServerPulse.Backend/AnalyzerApi.Core/Dtos/Endpoints/Analyze/GetSlotStatistics/GetSlotStatisticsResponse.cs
using AnalyzerApi.Core.Dtos.Responses.Events;
using AnalyzerApi.Core.Dtos.Responses.Statistics;

namespace AnalyzerApi.Core.Dtos.Endpoints.Analyze.GetSlotStatistics
{
    public class GetSlotStatisticsResponse
    {
        public DateTime CollectedDateUTC { get; set; }
        public ServerLifecycleStatisticsResponse? GeneralStatistics { get; set; }
        public ServerLoadStatisticsResponse? LoadStatistics { get; set; }
        public ServerCustomStatisticsResponse? CustomEventStatistics { get; set; }
        public IEnumerable<LoadEventResponse> LastLoadEvents { get; set; } = [];
        public IEnumerable<CustomEventResponse> LastCustomEvents { get; set; } = [];
    }
}
